documentation/generateDocs.js

```diff
@@ -0,0 +1,275 @@
+                var sub_folders = path.join(directoryPath, x);
+                fs.readdirSync(sub_folders).forEach(file => {
+                    mdFiles[x].push(file);
+                  });
+            }
+            var search_md_file = words[i].longname.split(".")[1] + '.md';
+            // checking md files inside core
+            if(words[i].longname.split(".")[0] == "cy")
+            {
+                if(mdFiles["cy"].indexOf(search_md_file) != -1)
+                {
+                    func.md = "cy/" + words[i].longname.split(".")[1];
+                }
+            }
+            // checking md files inside animation
+            else if(words[i].longname.split(".")[0] == "ani")
+            {
+                if(mdFiles["ani"].indexOf(search_md_file) != -1)
+                {
+                    func.md = "ani/" + words[i].longname.split(".")[1];
+                }
+            }
+            // checking md files inside layout
+            else if(words[i].longname.split(".")[0] == "layout")
+            {
+                if(mdFiles["layout"].indexOf(search_md_file) != -1)
+                {
+                    func.md = "layout/" + words[i].longname.split(".")[1];
+                }
+            }
+            // checking md files inside collection
+            else
+            {
+                if(mdFiles["collection"].indexOf(search_md_file) != -1)
+                {
+                    func.md = "collection/" + words[i].longname.split(".")[1];
+                }
+            }
+            
+            fns.push(func);
+        }
+    }
+    
+}
+
+// save generated file
+fs.writeFile ( path.join(__dirname, "./jsdocAnnotations.json"), JSON.stringify(fns, null, 4), function(err) {
+    if (err) throw err;
+    console.log('complete');
+});
+
+var mappings=fns;
+
+for(var i in dockmaker_elements.sections)
+{
+    if(dockmaker_elements.sections[i].name != undefined && ( dockmaker_elements.sections[i].name == 'Core' || dockmaker_elements.sections[i].name == 'Collection' || dockmaker_elements.sections[i].name == 'Animations' || dockmaker_elements.sections[i].name == 'Layouts' ))
+    {
+        for( var j in dockmaker_elements.sections[i].sections)
+        {
+            if(dockmaker_elements.sections[i].sections[j].fns != undefined)
+            {
+                for( var k in dockmaker_elements.sections[i].sections[j].fns)
+                {
+                    for( var x in mappings )
+                    {
+                        if(dockmaker_elements.sections[i].sections[j].fns[k].name == mappings[x].name)
+                        {
+                            dockmaker_elements.sections[i].sections[j].fns[k] = mappings[x];
+                            break;
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+
+// save generated file
+fs.writeFile ( path.join(__dirname, "./docmaker.json"), JSON.stringify(dockmaker_elements, null,4), function(err) {
```

Comment:
If you use `fs.writeFileSync()` everywhere you use `fs.writeFile()`, you won't need the sleep delay.

Reply:
Fixed.